crates/pipeline_manager/migrations/V6__add_external_service.sql

```diff
@@ -0,0 +1,20 @@
+
+-- External service
+-- It represents a service (such as MySQL, Postgres, etc.) that a connector can refer to in
+-- its config. A connector can refer to zero, one or multiple external services.
+-- External services are declared separately to reduce duplication and to make it
+-- easier to create connectors. An external service has its own configuration, which
+-- generally includes hostname, port, access credentials, and any service parameters.
+CREATE TABLE IF NOT EXISTS external_service (
+    id uuid PRIMARY KEY,                  -- Unique identifier (used primarily for foreign key relations)
+    tenant_id uuid NOT NULL,              -- Tenant the external service belongs to
+    name varchar NOT NULL,                -- Immutable unique name given by the tenant (e.g., mysql-example) used to refer to it in the API
```

Comment:
Is it already immutable in our API?

Reply:
Yes, the PATCH `/external_services/id` endpoint only allows updating description and config.

Reply:
We could also enforce it in the database with a trigger potentially (?)